test/infrastructure/docker/api/v1beta1/dockercluster_types.go

```diff
@@ -35,7 +35,7 @@ type DockerClusterSpec struct {
 
 	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
 	// +optional
-	ControlPlaneEndpoint APIEndpoint `json:"controlPlaneEndpoint"`
+	ControlPlaneEndpoint APIEndpoint `json:"controlPlaneEndpoint,omitempty"`
```

Comment:
I'm +1 to omit empty here and on the nested fields, but we usually keep this field consistent with Cluster in core Cluster API
@sbueringer opinions?

Reply:
I think I wouldn't change it in this PR.

We have an issue open to make the field a pointer in the Cluster struct and we deferred that one to the next apiVersion.

Not sure exactly what the impact is of adding omitempty here without making the field a pointer, but I would probably defer changes like that to the next apiVersion (#6416).

(same applies to the nested fields)

Reply:
@alexander-demicev Do we have to make those changes to the fields to make this PR work?

Reply:
Additional info. Not having omitempty is something that we also have in infra providers. We explicilty added handling in the topology controller to handle zero values of host and port correctly. I think by adding omitempty to CAPD we are losing test coverage for this case.

Reply:
I can remove omitempty tag, it doesn't block the PR